lib/backend/dynamo/configure.go

```diff
@@ -42,137 +41,80 @@ func (b *Backend) setContinuousBackups(ctx context.Context) error {
-	})
-	if err != nil {
+	}); err != nil {
 		return convertError(err)
 	}
 
 	// Define scaling policy. Defines the ratio of {read,write} consumed capacity to
 	// provisioned capacity DynamoDB will try and maintain.
-	_, err = svc.PutScalingPolicy(&applicationautoscaling.PutScalingPolicyInput{
-		PolicyName:        aws.String(fmt.Sprintf("%v-%v", b.TableName, readScalingPolicySuffix)),
+	if _, err := svc.PutScalingPolicy(&applicationautoscaling.PutScalingPolicyInput{
+		PolicyName:        aws.String(fmt.Sprintf("%v-%v", tableName, readScalingPolicySuffix)),
 		PolicyType:        aws.String(applicationautoscaling.PolicyTypeTargetTrackingScaling),
-		ResourceId:        aws.String(fmt.Sprintf("%v/%v", resourcePrefix, b.TableName)),
+		ResourceId:        aws.String(fmt.Sprintf("%v/%v", resourcePrefix, tableName)),
 		ScalableDimension: aws.String(applicationautoscaling.ScalableDimensionDynamodbTableReadCapacityUnits),
 		ServiceNamespace:  aws.String(applicationautoscaling.ServiceNamespaceDynamodb),
 		TargetTrackingScalingPolicyConfiguration: &applicationautoscaling.TargetTrackingScalingPolicyConfiguration{
 			PredefinedMetricSpecification: &applicationautoscaling.PredefinedMetricSpecification{
 				PredefinedMetricType: aws.String(applicationautoscaling.MetricTypeDynamoDbreadCapacityUtilization),
 			},
-			TargetValue: aws.Float64(b.Config.ReadTargetValue),
+			TargetValue: aws.Float64(params.ReadTargetValue),
 		},
-	})
-	if err != nil {
+	}); err != nil {
 		return convertError(err)
 	}
-	_, err = svc.PutScalingPolicy(&applicationautoscaling.PutScalingPolicyInput{
-		PolicyName:        aws.String(fmt.Sprintf("%v-%v", b.TableName, writeScalingPolicySuffix)),
+	if _, err := svc.PutScalingPolicy(&applicationautoscaling.PutScalingPolicyInput{
+		PolicyName:        aws.String(fmt.Sprintf("%v-%v", tableName, writeScalingPolicySuffix)),
 		PolicyType:        aws.String(applicationautoscaling.PolicyTypeTargetTrackingScaling),
-		ResourceId:        aws.String(fmt.Sprintf("%v/%v", resourcePrefix, b.TableName)),
+		ResourceId:        aws.String(fmt.Sprintf("%v/%v", resourcePrefix, tableName)),
 		ScalableDimension: aws.String(applicationautoscaling.ScalableDimensionDynamodbTableWriteCapacityUnits),
 		ServiceNamespace:  aws.String(applicationautoscaling.ServiceNamespaceDynamodb),
 		TargetTrackingScalingPolicyConfiguration: &applicationautoscaling.TargetTrackingScalingPolicyConfiguration{
 			PredefinedMetricSpecification: &applicationautoscaling.PredefinedMetricSpecification{
 				PredefinedMetricType: aws.String(applicationautoscaling.MetricTypeDynamoDbwriteCapacityUtilization),
 			},
-			TargetValue: aws.Float64(b.Config.WriteTargetValue),
+			TargetValue: aws.Float64(params.WriteTargetValue),
 		},
-	})
-	if err != nil {
+	}); err != nil {
 		return convertError(err)
 	}
 
 	return nil
 }
 
-// disableAutoScaling disables auto scaling.
-func (b *Backend) disableAutoScaling(ctx context.Context) error {
```

Comment:
So once enabled, there's no way to disable it anymore? Or is the "set" method above with zero values does equivalent thing?

Reply:
Good question.

In the initial implementation it was implemented if `auto_scaling: false` it would disable it. This had the unintended consequence that not setting it would cause Teleport to disable auto scaling. This isn't ideal because if someone is configuring DynamoDB elsewhere, Teleport could be disabling your auto scaling settings.

I could have made the bool a pointer but it just felt like it was getting too complex at that point.